### PR TITLE
Add GitHub Action workflows

### PR DESCRIPTION
**Pull request by fkondej**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+jobs:
+
+  build:
+    name: Build on ${{ matrix.os }} with Go ${{ matrix.go }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
+        go: ['1.15', '1.14']
```
Comment (ashleyvega) [inline .github/workflows/go.yml:22]: Please add `1.16` and remove `1.14`. The official supported Golang versions are always only "the current one" and "the previous one".

File: .github/workflows/yml-lint.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (ashleyvega) [inline .github/workflows/yml-lint.yml:1]: Please rename file to `yamllint.yml` to match other repos.

File: .github/workflows/yml-lint.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+name: YAML lint
+
+on:
+  pull_request:
+    branches: [main, develop]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
```
Comment (ashleyvega) [inline .github/workflows/yml-lint.yml:9]: Please add `null` as the explicit value for `workflow_dispatch`.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    ifneq ($(RELEASE_VERSION),)
+        VERSION := $(RELEASE_VERSION)
+    else
+        # No tag, so make one
+        VERSION := $(shell git describe --tags 2>/dev/null)
+    endif
+    VERSION_HASH := $(shell echo "$(GITHUB_SHA)" | cut -b1-8)
+endif
+
+GO_FLAGS := -ldflags "-X main.Version=$(VERSION) -X main.VersionHash=$(VERSION_HASH)"
```
Comment (ashleyvega) [inline Makefile:16]: Please remove anything version-related, since it's not used in the Go source.

**Review by ashleyvega — CHANGES_REQUESTED** [review]
minor changes requested.

File: main_test.go [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package main //nolint:testpackage
```
[the remark this answers is not in the document]
Comment (fkondej) [reply main_test.go:1]: leftover - will remove it

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**